appinfo/app.php

```diff
@@ -22,16 +22,12 @@
  *
  */
 
- $eventDispatcher = \OC::$server->getEventDispatcher();
- $eventDispatcher->addListener(
- 	'OCA\Files::loadAdditionalScripts',
- 	function() {
- 		\OCP\Util::addScript('files_3d', 'files3d');
- 	}
- );
+namespace OCA\Files3d\AppInfo;
 
 use OC\Files\Type\Detection;
 
+\OC::$server->query(Application::class);
```

Comment:
```suggestion
$app = \OC::$server->query(Application::class);
$app->register()
```

And then you create a dedicated register function in your `Application.php` where you do all your app init :)
That is the clean way

That way you can inject the MimeTypeDetector service in the constructor and use it to register the new Types like this

`$this->mimeTypeDetector->registerType('dae', 'model/vnd.collada+xml', null);`